zellij-utils/src/setup.rs

```diff
@@ -376,16 +377,16 @@ impl Setup {
 
         Ok(())
     }
-    fn generate_completion(shell: String) {
-        let shell = match shell.parse() {
+    fn generate_completion(shell: &str) {
+        let shell: Shell = match shell.parse() {
```

Comment:
Since the `Shell` can only be in lower-case, it would be nice to have a part to convert it to lower-case.

```rust
let shell: Shell = match.shell.to_lowercase().parse() {
...
}
```

Reply:
I wanted to automate this `parse` away by using `ArgEnum`, but since the `generate_completion` is a flag in a `Serializable` struct, it won't work...

It seems this used to be a separate subcommand before #503, is there any reason it was moved to a flag?

Reply:
Hmm... I don't know the history of this, maybe @a-kenji knows..?

Reply:
Ah, but AFAICS all the cli-related structs are serialized and sent to client, so `clap_complete::Shell` couldn't be used as an argument anyway without some redesign, so I'll just stick with this change for now 😅